charms/worker/k8s/src/charm.py

```diff
@@ -351,11 +351,11 @@ def _revoke_cluster_tokens(self, event: ops.EventBase):
 
         """
         log.info("Garbage collect cluster tokens")
-        to_remove = set()
+        to_remove = None
         if self.is_dying:
-            to_remove |= {self.unit}
-        if unit := _cluster_departing_unit(event):
-            to_remove |= {unit}
+            to_remove = self.unit
+        elif unit := _cluster_departing_unit(event):
+            to_remove = unit
```

Comment:
let's just simplify this.  If we're handling a hook b/c this unit is dying... only revoke the cluster tokens for the node specified.  This code is only run on the control-plane leader.  If a non "leaving style" event is handled `to_remove` its empty.  If it's a leaving-style event, then it's the unit that is leaving.  It could be a worker leaving, it could be a control-plane leaving -- and HARDEST -- it could be the lead control-plane leaving